drivers/spi/spi_nrfx_spim.c

```diff
@@ -27,9 +27,6 @@ LOG_MODULE_REGISTER(spi_nrfx_spim, CONFIG_SPI_LOG_LEVEL);
 #define SPI_BUFFER_IN_RAM 1
 #endif
 
-/* Maximum chunk length (depends on the EasyDMA bits, equal for all instances) */
-#define MAX_CHUNK_LEN BIT_MASK(SPIM0_EASYDMA_MAXCNT_SIZE)
```

Comment:
As far as I know, this is SoC specific, not instance specific. So this needs to be fixed by providing something like `SPIM_EASYDMA_MAXCNT_SIZE` instead.

Reply:
I think that any solution, either HAL providing non-instance specific define or driver defining this value on its own based on SoC, needs to minimize ROM usage, ie, not add artificial instance specific prop because of how HAL provides such value.

Reply:
If that's the case, then Devicetree needs to be used and capture it per-instance, not HAL definitions. Remember that our reliance on peripheral numbers in drivers will need to go sooner than later, we're the only vendor doing this, resulting in unnecessarily complicated Kconfig hackery.

Reply:
Using one SPIM instance specific setting to set it to the all used instances in design error. We really do not care about this potential 20B of flash that we can spare because of this reverted commit. Please keep in mind that you would spare 4B x Number of used SPIM instances in a price of creating obscured code that starts to be fragile. We have defined _EASYDMA_MAXCNT_SIZE for each instance and we have to use it.
The original commit blocks our work and generates an risk of not providing the code on time!
If you really obtain on sparing this 4B of memory per instance, provide any other solution that does not rely on some specific instance. Also - in such situation - provide build time assertion that the _EASYDMA_MAXCNT_SIZE is not different on the specific instance that the one defined globally.

Reply:
FYI I've opened https://github.com/zephyrproject-rtos/zephyr/pull/54663